### PR TITLE
Issue 475 multiple ttl monitors

### PR DESCRIPTION
**Pull request by mortenjc**
### Issue reference / description

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

File: src/modules/ttlmonitor/TTLMonitorBase.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  TTLMonitorBase(BaseSettings const &settings, struct TTLMonitorSettings &LocalTTLMonitorSettings);
+  ~TTLMonitorBase() = default;
+
+  void input_thread();
+  void processing_thread();
+
+protected:
+  struct Counters Counters{};
+  TTLMonitorSettings TTLMonitorModuleSettings;
+  EV42Serializer *Serializer;
```
Comment (JennyLouise) [inline src/modules/ttlmonitor/TTLMonitorBase.h:35]: Check if using this serializer

File: src/modules/ttlmonitor/TTLMonitorInstrument.cpp [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  for (const auto & readout : VMMParser.Result) {
+
+    if (DumpFile) {
+      dumpReadoutToFile(readout);
+    }
+
+    XTRACE(DATA, DEB, "readout: RingId %d, FENId %d, VMM %d, Channel %d, TimeLow %d",
+           readout.RingId, readout.FENId, readout.VMM, readout.Channel, readout.TimeLow);
+
+    if (readout.RingId/2 != 11) {
```
Comment (JennyLouise) [inline src/modules/ttlmonitor/TTLMonitorInstrument.cpp:66]: Consider making ringid and fenid configurable

File: src/modules/ttlmonitor/test/TTLMonitorBaseTest.cpp [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    LocalSettings.ConfigFile = "TTLMonitor.json";
+    Settings.RxSocketBufferSize = 100000;
+    Settings.NoHwCheck = true;
+  }
+  void TearDown() override {}
+
+  BaseSettings Settings;
+  TTLMonitor::TTLMonitorSettings LocalSettings;
+};
+
```
Comment (JennyLouise) [inline src/modules/ttlmonitor/test/TTLMonitorBaseTest.cpp:77]: run clang-format

File: src/modules/ttlmonitor/test/TTLMonitorInstrumentTest.cpp [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+TEST_F(TTLMonitorInstrumentTest, Constructor) {
+  ASSERT_EQ(counters.RingCfgErrors, 0);
+  ASSERT_EQ(counters.FENCfgErrors, 0);
+}
+
+
+TEST_F(TTLMonitorInstrumentTest, BeamMonitor) {
+  makeHeader(freia->ESSReadoutParser.Packet, MonitorReadout);
+
+  freia->VMMParser.setMonitor(true);
```
Comment (JennyLouise) [inline src/modules/ttlmonitor/test/TTLMonitorInstrumentTest.cpp:158]: change name from freia to ttlmonitor

**Review by JennyLouise — CHANGES_REQUESTED** [review]

File: src/modules/ttlmonitor/TTLMonitorBase.h (hunk 1 above)
Comment (mortenjc) [reply src/modules/ttlmonitor/TTLMonitorBase.h:35]: this is okj.

File: src/modules/ttlmonitor/TTLMonitorInstrument.cpp (hunk 2 above)
Comment (mortenjc) [reply src/modules/ttlmonitor/TTLMonitorInstrument.cpp:66]: done

File: src/modules/ttlmonitor/test/TTLMonitorInstrumentTest.cpp (hunk 4 above)
Comment (mortenjc) [reply src/modules/ttlmonitor/test/TTLMonitorInstrumentTest.cpp:158]: done

File: src/modules/ttlmonitor/test/TTLMonitorBaseTest.cpp (hunk 3 above)
Comment (mortenjc) [reply src/modules/ttlmonitor/test/TTLMonitorBaseTest.cpp:77]: will do

**Review by JennyLouise — APPROVED** [review]

**Pull request merged**